### ISSUE TITLE
✨ API - TalentNominationGroup

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#12296]
## ✨ Feature

Build the API for the talent event candidate model that represents a person nominated one or more times for an event.

## 🕵️ Details

We need one new model:

1. TalentNominationGroup - represents an employee who has been nominated one or more times for an event

We need one new mutation:

1. UpdateTalentNominationGroup

Additionally, we need to update `SubmitTalentNomination` to create a TalentNominationGroup for the nomination if it doesn't exist yet and attach the nomination to it.

## 🎨 Design

@JoshBeveridge 

https://www.figma.com/design/N9cykAUckUTgf92YUxzGDO/Events-and-nominations-(Communities)?node-id=151-8&node-type=canvas&t=LdL7KkRzqF7Yikh4-0

## 📸 Screenshot

![Image](https://github.com/user-attachments/assets/6c01f667-bc8a-495a-8350-7d8cbdd62c7c)

## 🙋‍♀️ Proposed Implementation

```graphql
type TalentNominationGroup {
   created_at: Date
   nominee: User! @belongsTo
   talentNominationEvent: TalentNominationEvent! @belongsTo
   advancement_decision: Decision (Approved, Rejected)
   advancement_reference_confirmed: Boolean
   advancement_notes: String
   lateral_movement_decision: Decision (Approved, Rejected)
   lateral_movement_notes: String
   dev_program_decision: Decision (Approved, Rejected)
   dev_program_notes: String
   computed_status: Status (Approved, Rejected, Partially Approved)
   talentNominations: [TalentNomination!] @hasMany
}
```

## ✅ Acceptance Criteria

- [ ] Database unique constraint: only one TalentNominationGroup per nominee and event
- [ ] When a nomination is submitted, it is attached to a new or existing TalentNominationGroup
- [ ] community talent coordinator can query for a list of all TalentNominationGroup in any event in their community
- [ ] community talent coordinator can query for a single TalentNominationGroup by ID
- [ ] community talent coordinator can query for a list of TalentNominations attached to a TalentNominationGroup
- [ ] `computes_status` is calculated, not updateable by users
- [ ] All model changes (including relationships) tracked in the activity_log table
- [ ] local env has seeded rows

Out of scope:
- comments
- bookmarks

## 🛑 Blockers

```[tasklist]
### Blocked By
- [ ] #12293
```
